content/knowledge_base/flare/consensus/avalanche/specification.md

```diff
@@ -0,0 +1,210 @@
+Since conflicts are [transitive](../../../glossary.md#Transitive-Relation), if PTi and PTj are conflicting they belong to the same conflict set of transactions — PTi = PTj.
+Conflicting transation have the equivalence relation because they are equivocations spending the same UTXO.
+
+The parent connection to T' from T is expressed as `T' <— T`.
+The relation `T' *<— T` is its reflexive [transitive closure](../../../glossary.md#Transitive-Closuse-of-a-Graph) meaning that there is path from T to T'.
+If T' <— T, then every node in the system that has T will also have T' and will know about the same relation `T' <— T`.
+If such a relation between these transactions do not exist, then no node will end up with T' <— T.
+
+DAGs built in different nodes are guaranteed to be compatible, though at any moment in time they might not have a complete view of all vertices in the system.
+Each node can compute a confidence value `d U (T)` (confidence value for node `u` about transaction `T`) from the progeny as the sum of transaction T decendants cihts from the progeny and its own chit.
+Each transaction initially has a chit of 0 before the node gets the query results.
+If the node collects a threshold of α 'yes' votes from the query the value of transaction T chit is set to 1, otherwise it will stay foerever 0.
+A chit value reflect the result from a single query of its associated transaction and becomes immutable afterwards, while confidence `d` can increase as the DAG grows by collecting more chits in its progeny.
+Additionally each node maintains its own local list of known nodes: Nu — nodes known to `u` is a subset of the set of all existing nodes: `Nu ⊆ N`.
+
+Each node implements an event-driven state machine, centered around a query that serves both to solicit votes on each transaction and to notify other nodes of the existence of newly discovered transactions.
+When a node `u` discovers a trancation `T` through a query, it starts a one-time query process by sampling k random peers and sending a message to them, after T is delivered to the node by onReceiveTx.
+Node `u` answers a query by checking whether each T' such that `T' *<— T` is currently prefered among competing transactions `∀T'' ∈ PT'` from conflict set P of T'.
+If every single ancestor T' fulfills this criterion, the transation is said to br strongly preferred and receives a yes-vote (1).
+A failure on this criterion at any T' yields a no-vote (0).
+When `u` accumulates k responses, it checks whether there are α yes-votes for T, and if so grants the chit value (cT = 1).
+This process will yield a labeling of the DAG with a chit value and associated confidence for each transaction T.
+
+Example of (chit,confidence) values:
+
+```text
+
+T1(1,6)
+├── T2(1,5)
+│    ├── T4(1,2) ── T8(1,1)
+│    │               │
+│    └────────────── T5(1,3)
+│                    └──── T9(1,1)
+│
+└── T3(0,0)
+    ├── T6(0,0)
+    └── T7(0,0)
+
+```
```

Comment:
Is there a need for the leading/trailing newlines in this code block?

Reply:
No, I'll remove them, thanks!

Reply:
Removed the newlines https://github.com/optakt/optakt.github.io/pull/13/commits/e0e27e6be8c0aef4153af5e2add9722380c23a59#diff-2d4f16e70781857aec6689a748ad155e45611b81f184aee076bb139890410b98L75